octavia-cli/octavia_cli/generate/renderer.py

```diff
@@ -0,0 +1,161 @@
+            return []
+        one_of_values = []
+        for one_of_value in self.oneOf:
+            properties = get_object_fields(one_of_value)
+            one_of_values.append(properties)
+        return one_of_values
+
+    def _get_array_items(self) -> List["FieldToRender"]:
+        """If the field is an array of objects, retrieve fields of these objects.
+
+        Returns:
+            [list]: List of fields
+        """
+        if self._is_array_of_objects:
+            required_fields = self.items.get("required", [])
+            return parse_fields(required_fields, self.items["properties"])
+        return []
+
+    def _get_required_comment(self) -> str:
+        return "REQUIRED" if self.required else "OPTIONAL"
+
+    def _get_type_comment(self) -> str:
+        return self.type if self.type else None
+
+    def _get_secret_comment(self) -> str:
+        return "🤫" if self.airbyte_secret else None
+
+    def _get_description_comment(self) -> str:
+        return self.description if self.description else None
+
+    def _get_example_comment(self) -> str:
+        example_comment = None
+        if self.examples:
+            if isinstance(self.examples, list):
+                if len(self.examples) > 1:
+                    example_comment = f"Examples: {', '.join([str(example) for example in self.examples])}"
+                else:
+                    example_comment = f"Example: {self.examples[0]}"
+            else:
+                example_comment = f"Example: {self.examples}"
+        return example_comment
+
+    def _get_default(self) -> str:
+        if self.const:
+            return self.const
+        return self.default
+
+    @staticmethod
+    def _build_comment(comment_functions: Callable) -> str:
+        return " | ".join(filter(None, [comment_fn() for comment_fn in comment_functions])).replace("\n", "")
+
+
+def parse_fields(required_fields: List[str], fields: dict) -> List["FieldToRender"]:
+    return [FieldToRender(f_name, f_name in required_fields, f_metadata) for f_name, f_metadata in fields.items()]
+
+
+def get_object_fields(field_metadata: dict) -> List["FieldToRender"]:
+    if field_metadata.get("properties"):
+        required_fields = field_metadata.get("required", [])
+        return parse_fields(required_fields, field_metadata["properties"])
+    return []
+
+
+class ConnectionSpecificationRenderer:
+    TEMPLATE = JINJA_ENV.get_template("source_or_destination.yaml.j2")
+
+    def __init__(self, resource_name: str, definition: BaseDefinition) -> None:
+        self.resource_name = resource_name
+        self.definition = definition
+
+    def _parse_connection_specification(self, schema: dict) -> List[List["FieldToRender"]]:
+        if schema.get("oneOf"):
+            roots = []
+            for one_of_value in schema.get("oneOf"):
+                required_fields = one_of_value.get("required", [])
+                roots.append(parse_fields(required_fields, one_of_value["properties"]))
```

Comment:
Isn't it possible that there could be another `oneOf` field nested inside a value of a parent `oneOf` field? If so, would this logic handle that case properly? It seems like we might need some recursion to be able to handle that properly, right?

Reply:
You're right, I did not implement recursion at first for the two following reasons:
* Might make the Jinja template a bit more complex.
* I did not find connectors for which the spec has nested oneof fields. I need to double check this assertion.

I'll evaluate what it takes to implement recursion to support nested oneof.

Reply:
destination-s3 has a nested oneof field 😢

Reply:
In this function, I only try to handle the case when the root of the schema is a `oneof` field (e.g the `destination-e2e-test` connector). 
But you're right, the current jinja templates did not handle nested `oneOf` / `arrays` / `object`. I fixed it in https://github.com/airbytehq/airbyte/pull/10132/commits/c6a084a906b0e5b875d2c572cd9cf61f00c0dc5a : I leveraged Jinja macro recursion and implemented an integration test for a connector with nested `oneOf` field (`destination-s3`).